lib/Parser/Parse.cpp

```diff
@@ -5811,7 +5815,7 @@ void Parser::ParseTopLevelDeferredFunc(ParseNodeFnc * pnodeFnc, ParseNodeFnc * p
     {
         ParseExpressionLambdaBody<false>(pnodeFnc);
     }
```

Comment:
Does checking HasDefaultArguments tell us what we need to know? We want to skip the children of a function with default arguments, just not the ones that are *in* default arguments, i.e., in the param scope.

Reply:
This is checking `pnodeFncParent->HasDefaultArguments()` so we should be parsing a function nested in a parent which has default arguments, right? This is a little broad as the parent might have other functions not in the param scope but I wanted to be safer here and return when we can enable this for all functions nested in a function with default arguments.

Reply:
It's quite broad, yes. It's going to disable the optimization for a lot of functions that could benefit. It's not unsafe to check in this way, but you're going to want to put the right logic here ASAP. Please open an issue so it doesn't get lost.

Reply:
Agreed, I opened https://github.com/Microsoft/ChakraCore/issues/5156 to track it.